examples/recording_service/leveldb-integration/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,180 @@
+#
+# (c) 2019 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+#
+#  RTI grants Licensee a license to use, modify, compile, and create derivative
+#  works of the Software.  Licensee has the right to distribute object form
+#  only for use with RTI products.  The Software is provided "as is", with no
+#  warranty of any type, including any warranty for fitness for any purpose.
+#  RTI is under no obligation to maintain or support the Software.  RTI shall
+#  not be liable for any incidental or consequential damages arising out of the
+#  use or inability to use the software.
+#
+cmake_minimum_required(VERSION 3.11)
+project (LevelDbIntegration)
+
+set(CMAKE_DEBUG_POSTFIX "d")
+
+# Find RTI Connext dependencies
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CONNEXTDDS_DIR}/resource/cmake")
+
+find_package(
+    RTIConnextDDS "6.0.1"
+    REQUIRED
+    COMPONENTS
+        core
+        routing_service
+)
+
+set(LEVELDB_LIB_NAME "leveldb")
+
+find_library(
+    leveldb_lib
+    NAMES ${LEVELDB_LIB_NAME}
+    PATHS "${LEVELDB_DIR}"
+)
+
+# Disable RTTI.
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    string(REGEX REPLACE "/GR" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /GR-")
+else()
+    string(REGEX REPLACE "-frtti" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti")
+endif()
+
+# Enable sanitizers
+if(ENABLE_SANITIZERS)
```

Comment:
@iblancasa what do you think about this? is that something we should embed somewhere else? if not, is this the right way to do it?

Reply:
It is